mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -281,16 +280,17 @@ class Word2Vec extends Serializable with Logging {
     val expTable = sc.broadcast(createExpTable())
     val bcVocab = sc.broadcast(vocab)
     val bcVocabHash = sc.broadcast(vocabHash)
-
-    val sentences: RDD[Array[Int]] = words.mapPartitions { iter =>
+    //each partition is a collection of sentences, will be translated into arrays of Index integer
+    val sentences: RDD[Array[Int]] = dataset.mapPartitions { iter =>
```

Comment:
Agreed. Will do.
Btw: the usage of maxsentencelenth in  the python code is different than the original mllib word2vec. It's actually respect sentence structure on condition that it's shorter than the max allowance, which is a mixture of my change with the adress of @srowen too long sentence concern. But as you mentioned and I mentioned below, the too long sentence should not be a concern in normal situation. The only exception I can think of is to use a poor sentence splitter plus using concatenated large portion of a big corpus as one individual document in the input.